### DISCUSSION TITLE
how to use Cython to build fastapi and pydantic project

### DISCUSSION THREAD

**Discussion by panla** [fastapi/fastapi#6574] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.get('/api/index/1')
def get(token: str = Header(..., description='token')):
    return {"success": True}
```

### Description

after use Cython to transform it to .so

run it will raise

- TypeError: Expected unicode, got Header

### Operating System

Linux

### Operating System Details

deepin

### FastAPI Version

0.68

### Python Version

3.8.11

### Additional Context

- Cython: 3.0.0a9
- pydantic: 1.8.2

**Comment by labeneator** [comment]
Hi @panla,
 How were you able to resolve this issue?

**Comment by panla** [comment]
> Hi @panla, How were you able to resolve this issue?

No, Use Cython, we can translation simple code, but can`t deal complex code, 
and, we can not rewrite exists code to accommodate Cython.

**Reply by ghost** [reply]
Has this issue been resolved with any walkaround?

**Comment by panla** [comment, marked as answer]
https://github.com/panla/cy_encrypt

**Comment by FFFro** [comment]
TypeError: Expected unicode, got Header 大佬这个问题解决了吗

**Reply by panla** [reply]
## 代码

tools.py

```python
from setuptools import setup

from Cython.Build import cythonize

# Cython 编译参数
COMPILER_DIRECTIVES = {
    'language_level': 3,
    'always_allow_keywords': True,
    'annotation_typing': False
}


class Operator:

    def compile(self):
        names = ['demo/application.py']

        # 执行
        setup(
            ext_modules=cythonize(names, quiet=True, compiler_directives=COMPILER_DIRECTIVES),
            script_args=['build_ext', '--inplace']
        )

    def execute(self):
        self.compile()


Operator().execute()

```

application.py

```python
from fastapi import FastAPI
from fastapi import Header

app = FastAPI()


@app.get('/api')
def get(token: str = Header(..., description='token')):
    print(token)
    return {"success": True}

```

server.py

```python
from application import app

```

## 文件目录结构

```text
demo/
    application.py
tests/
    server.py
    application.so
tools.py
```

## 流程与测试

```text
在根目录执行 python tools.py
此时找到生成的 .so 文件
复制到 tests 目录下
在 tests 目录下执行 uvicorn server:app --reload

此时默认情况下, 会启动一个 监听 127.0.0.1:8000 的服务

使用 postman or apifox 等测试工具，访问 GET http://127.0.0.1:8000/api 在请求头中带上 token 参数 来做测试
检查效果
```

你这样看下效果行不行, 可能和 compiler_directives 参数有关系

**Reply by FFFro** [reply]
谢谢大佬，确实是compiler_directives 参数的问题。ds让我加上 'annotation_typing': True ，搞了半天都不对，结果应该传 'annotation_typing': False
